Makefile

```diff
@@ -46,7 +46,11 @@ vet:
 
 # Run go test against code
 test:
-	go test ./...
+	go test ./apis/...
+
+# Run conformance tests against controller implementation
```

Comment:
How are integrator intended to run these? checkout the repo and run `make conformance`?

I am not sure how it works for other conformance tests, but it would be cool to just import it and write some code like
```go
func TestServiceApis(t *testing.T) {
   serviceapis.RunConformance(t)
}
```

Reply:
That would be very cool but I'm not sure how it would work in practice. These tests will need a Kubernetes cluster with the controller implementation running as well as the ability to make network requests to a provided Gateway. So far the only way I've seen that done is with a standalone test suite that runs in a provided cluster. I've only looked at k/k and ingress conformance tests though, I'm open to other approaches here.

Reply:
For my example code, we already have a full framework for deploying Istio (the controller implementation), connecting to k8s cluster, etc. Which I would imagine other implementations also have.

The tricky parts I think is that tests cannot be imported (so it needs to be a non-test, which is then imported by service-apis tests like above), and I think testdata/ will be broken so you would have to embed all the files I guess? Maybe other things would break as well, I really don't know...

Reply:
> These tests will need a Kubernetes cluster with the controller implementation running as well as the ability to make network requests to a provided Gateway. So far the only way I've seen that done is with a standalone test suite that runs in a provided cluster.

This is what I'm familiar with. It usually strikes a good balance between maintainability across projects and ease of running conformance tests within implementations' CI and a common CI that can track conformance across all implementations.

Reply:
It would be nice to also have the ability to opt out of tests. I do understand that makes it easy to be non-compliant, but there may be partial implementations or other features that a controller may not fully support (i.e. TCP) and would always fail the test run.

Reply:
Good point. We definitely need to be able to describe different levels of conformance, and ideally only run tests for the tests that are associated with those areas. One potential option would be to use the same kind of tags for tests that Kubernetes upstream uses.